static/js/reducers/index.js

```diff
@@ -1,24 +1,52 @@
+// @flow
 import { combineReducers } from 'redux';
 import {
-    UPDATE_CHECKBOX
+  FETCH_PROCESSING,
+  FETCH_SUCCESS,
+  FETCH_FAILURE,
+
+  REQUEST_PAYMENT,
+  RECEIVE_PAYMENT_SUCCESS,
+  RECEIVE_PAYMENT_FAILURE,
+  SET_TOTAL,
 } from '../actions';
+import type { Action } from '../flow/reduxTypes';
 
-const INITIAL_CHECKBOX_STATE = {
-  checked: false
+export type PaymentState = {
+  fetchStatus?: string,
 };
+const INITIAL_PAYMENT_STATE: PaymentState = {};
 
-export const checkbox = (state = INITIAL_CHECKBOX_STATE, action) => {
+export const payment = (state: PaymentState = INITIAL_PAYMENT_STATE, action: Action) => {
   switch (action.type) {
-  case UPDATE_CHECKBOX:
-    return Object.assign({}, state, {
-      checked: action.payload.checked
-    });
+  case REQUEST_PAYMENT:
+    return { ...state, fetchStatus: FETCH_PROCESSING };
+  case RECEIVE_PAYMENT_SUCCESS:
+    return { ...state, fetchStatus: FETCH_SUCCESS };
+  case RECEIVE_PAYMENT_FAILURE:
+    return { ...state, fetchStatus: FETCH_FAILURE };
+  default:
+    return state;
+  }
+};
+
+export type UIState = {
+  total: string,
+};
+const INITIAL_UI_STATE = {
+  total: ''
+};
+
+export const ui = (state: UIState = INITIAL_UI_STATE, action: Action) => {
+  switch (action.type) {
+  case SET_TOTAL:
+    return { ...state, total: action.payload };
   default:
     return state;
   }
 };
 
 export default combineReducers({
-  checkbox
+  payment,
+  ui,
 });
```

Comment:
I'm getting a strange error when installing redux-api:

    ➜  bootcamp-ecommerce git:(gs/payment) dwatch yarn add redux-api
    yarn add v0.19.1
    [1/4] Resolving packages...
    error An unexpected error occurred: "ENOENT: no such file or directory, open '/usr/local/lib/node_modules/yarn/node_modules/har-validator/lib/index.js'".
    info If you think this is a bug, please open a bug report with the information provided in "/src/yarn-error.log".
    info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
    ➜  bootcamp-ecommerce git:(gs/payment)